### PR TITLE
Add sidebar 0.1 support for nested menu and guard behind experiment

### PR DESCRIPTION
**Pull request by leafsy**
Add the logic for nested menu lazy loading to `<amp-sidebar>` v0.1 and guard it behind the `amp-nested-menu` experiment.

### CODE REVIEW

File: extensions/amp-nested-menu/0.1/test/test-amp-nested-menu.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ describes.realWin(
       clock = lolex.install({target: win});
 
       // TODO(#25022): remove this toggle when cleaning up experiment post launch.
```
Comment (cathyxz) [inline extensions/amp-nested-menu/0.1/test/test-amp-nested-menu.js]: The issue number here should also be updated to https://github.com/ampproject/amphtml/issues/25343.

**Comment by cathyxz** [comment]
@rcebulko do you know why we're still getting this from owners bot? 
>  [NEEDS APPROVAL] tools/experiments/experiments-config.js (requested: kristoferbaxter)

I thought we fixed this with https://github.com/ampproject/amphtml/pull/25169.

**Comment by rcebulko** [comment]
> @rcebulko do you know why we're still getting this from owners bot?
> 
> > [NEEDS APPROVAL] tools/experiments/experiments-config.js (requested: kristoferbaxter)
> 
> I thought we fixed this with #25169.
> 
> I wonder if it's because child team members don't count as team members, so `working-groups` won't actually work properly for owners bot. Maybe it the proper owners group should be `reviewers` instead.

The rule in `tools/OWNERS`targets `experiment-config.js`, while this file is in `tools/experiments`. It looks like either you want the rule pattern to be `**/experiment-config.js` (if it should apply to all subdirectories), or that rule itself should move to `tools/experiments/OWNERS`

**Comment by cathyxz** [comment]
Thanks Ryan!

**Pull request merged**